experimenter/evaluate_pipeline_new.py

```diff
@@ -0,0 +1,199 @@
+    scoring_params=None,
+    scoring_random_seed: int=0):
+    """ 
+    Evaluate pipeline on problem using d3m's runtime cli. 
+    Wrapper function to execute d3m's runtime cli 'evaluate' command.
+    Arguments mirror the same arguments using the cli.
+    Only handles cases with a data preparation pipeline in the 
+    pipeline run.
+
+    Parameters
+    ----------
+    pipeline : path_like or uuid4 str
+        path to pipeline doc or pipeline ID
+    problem : path_like str
+        path to problem doc
+    input : path_like str
+        path to input full data
+    output_run : path_like str or '-'
+        path where pipeline_run doc
+        will be saved.
+        use '-' for stdin
+    random_seed : int
+        random seed to used for
+        pipeline run
+    data_pipeline_path: str
+        path to data prepation pipeline
+    data_random_seed: int
+        random_seed to be used in data preparation
+    data_params:
+        parameters for data preparation
+    scoring_params:
+        parameters for scoring pipeline
+    scoring_random_seed: int
+        random seed for scoring
+    scoring_pipeline: str
+        path to scoring pipeline
+    Return:
+    -------
+    None
+    
+    Raises:
+    -------
+    ValueError
+        when parameter value is
+        invalid
+    """    
+    if (not os.path.isfile(pipeline)):
+        raise ValueError('\'{}\' param not a file path'.format('pipeline'))
+
+    if (not os.path.isfile(problem)): 
+        raise ValueError('\'{}\' param not a file path'.format('problem'))
+
+    if (not os.path.isfile(input)):
+        raise ValueError('\'{}\' param not a file path'.format('input'))
+        
+    if (not os.path.isfile(data_pipeline_path)):
+        raise ValueError('\'{}\' param not a file path'.format('input'))
+        
+    if (not os.path.isfile(scoring_pipeline)):
+        raise ValueError('\'{}\' param not a file path'.format('input'))
+        
+    #TODO - call fit-score when the data pipeline is not defined in the pipeline run
+    args = ['d3m', 'runtime','--random-seed', str(random_seed), 'evaluate']
+    args.extend(('--pipeline', pipeline))
+    args.extend(('--problem', problem))
+    args.extend(('--input', input))
+    args.extend(('--output-run', output_run))
+    args.extend(('--data-pipeline', data_pipeline_path))
+    args.extend(('--data-random-seed', str(data_random_seed)))
+    args.extend(('--scoring-pipeline', scoring_pipeline))
+    args.extend(('--scoring-random-seed', str(scoring_random_seed)))
+    #add the data parameters to the cli arguments
+    if (data_params is not None):
+        for name, value in data_params.items():
+            args.extend(('--data-param', name, value))
+    #add the scoring parameters to the cli arguments
+    if (scoring_params is not None):
+        for name, value in scoring_params.items():
+            args.extend(('--scoring-param', name, value))
+    d3m_cli.main(args)
+    if (config.save_to_d3m is True):
```

Comment:
Make this an argument to the function, instead of accessing the config file directly. Accessing the config should be at the highest level functions.